sample-rv/start-rv.sh

```diff
@@ -0,0 +1,328 @@
+
+main() {
+    # Set environment variables only if not already set
+    export FDO_RV_DB="${FDO_RV_DB:-fdo_rv}"
+    export FDO_RV_DB_USER="${FDO_RV_DB_USER:-fdouser}"
+    export FDO_RV_DB_PASSWORD="${FDO_RV_DB_PASSWORD:-$(generateToken 30)}"
+    export FDO_RV_DB_PORT="${FDO_RV_DB_PORT:-5435}"
+    export POSTGRES_HOST_AUTH_METHOD="${POSTGRES_HOST_AUTH_METHOD:-scram-sha-256}"
+    export POSTGRES_IMAGE_TAG="${POSTGRES_IMAGE_TAG:-17}"
+    export HZN_DOCK_NET="${HZN_DOCK_NET:-hzn_horizonnet}"
+    export FIDO_DEVICE_ONBOARD_REL_VER=${FIDO_DEVICE_ONBOARD_REL_VER:-1.1.9} # https://github.com/fido-device-onboard/release-fidoiot/releases
+    export FDO_RV_SVC_AUTH=${FDO_RV_SVC_AUTH:-apiUser:$(generateToken 30)}
+    export FDO_SUPPORT_RELEASE=${FDO_SUPPORT_RELEASE:-https://github.com/fido-device-onboard/release-fidoiot/releases/download/v$FIDO_DEVICE_ONBOARD_REL_VER}
+    export FDO_RV_POSTGRES_CONTAINER="postgres-fdo-rv-service"
+    workingDir=fdo
+    deviceBinaryDir='pri-fidoiot-v'$FIDO_DEVICE_ONBOARD_REL_VER
+
+    check_root_user
+    create_and_switch_directory "$workingDir"
+    check_and_install_java
+    check_and_install_docker
+
+    # Start a DB container for FDO's rv service
+    docker rm -f $FDO_RV_POSTGRES_CONTAINER
+    docker run -d \
+        -e "POSTGRES_DB=$FDO_RV_DB" \
+        -e "POSTGRES_PASSWORD=$FDO_RV_DB_PASSWORD" \
+        -e "POSTGRES_USER=$FDO_RV_DB_USER" \
+        -e "POSTGRES_HOST_AUTH_METHOD=$POSTGRES_HOST_AUTH_METHOD" \
+        -e "POSTGRES_INITDB_ARGS=--auth-host=scram-sha-256 --auth-local=scram-sha-256" \
+        --health-cmd="pg_isready -U $FDO_RV_DB_USER" \
+        --health-interval=15s \
+        --health-retries=3 \
+        --health-timeout=5s \
+        --name $FDO_RV_POSTGRES_CONTAINER \
+        --network="$HZN_DOCK_NET" \
+        -p "$FDO_RV_DB_PORT":5432 \
+        postgres:"$POSTGRES_IMAGE_TAG"
+
+    download_and_extract_device_binaries
+    generate_and_copy_keys
+
+    api_user=$(echo "$FDO_RV_SVC_AUTH" | awk -F: '{print $1}')
+    api_password=$(echo "$FDO_RV_SVC_AUTH" | awk -F: '{print $2}')
+    # Replace db_user and db_password
+    sed -i -e "s/^db_user=.*/db_user=$FDO_RV_DB_USER/" "$PWD/$deviceBinaryDir/rv/service.env"
+    chk $? 'sed rv/service.env db_user'
+    sed -i -e "s/^db_password=.*/db_password=$FDO_RV_DB_PASSWORD/" "$PWD/$deviceBinaryDir/rv/service.env"
+    chk $? 'sed rv/service.env db_password'
+
+    # Configure rv/hibernate.cfg.xml to use PostgreSQL driver
+    sed -i -e "s/org.mariadb.jdbc.Driver/org.postgresql.Driver/" "$PWD/$deviceBinaryDir/rv/hibernate.cfg.xml"
+    chk $? 'sed rv/hibernate.cfg.xml driver_class'
+
+    # Replace hibernate.connection.username and password
+    sed -i -e "0,/hibernate.connection.username:/s|hibernate.connection.username:.*|hibernate.connection.username: $FDO_RV_DB_USER|" "$PWD/$deviceBinaryDir/rv/service.yml"
+    chk $? 'sed rv/service.yml hibernate.connection.username'
```

Comment:
how about wrapping it in a function to avoid duplication and enhance readability?